service/history/handler.go

```diff
@@ -681,6 +697,10 @@ func (h *handlerImpl) StartWorkflowExecution(
 
 	startRequest := wrappedRequest.StartRequest
 	workflowID := startRequest.GetWorkflowID()
+	if !h.workflowIDCache.AllowExternal(domainID, workflowID) {
```

Comment:
could you please add a unit test to see the method call. It is quite simple but the test would grow as we start to ratelimit .

Reply:
also a suggestion to skip the if like how we do in frontend.

https://github.com/uber/cadence/blob/master/service/frontend/api/handler.go#L873

Reply:
Added test and removed the if